### ISSUE TITLE
[BUG] RHEL 8.3 crc provides vague WARN when pdnsd service is running (Failed public DNS, Wildcard DNS)

### ISSUE THREAD

**Issue by dbibm** [code-ready/crc#2258]
### General information

  * OS: Linux (RHEL 8.3)
  * Hypervisor: KVM
  * Did you run `crc setup` before starting it (Yes)?
  * Running CRC on: Laptop

## CRC version
```bash
CodeReady Containers version: 1.25.0+0e5748c8
OpenShift version: 4.7.5 (embedded in executable)
```
  
## CRC status
```bash
CRC VM:          Running
OpenShift:       Unreachable (v4.7.5)
Disk Usage:      17.7GB of 32.74GB (Inside the CRC VM)
Cache Usage:     13.4GB
Cache Directory: /home/david.bensimon/.crc/cache
```

## CRC config
```bash
- consent-telemetry                     : no
```

## Host Operating System
```bash
NAME="Red Hat Enterprise Linux"
VERSION="8.3 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.3"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.3 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8.3:GA"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.3
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.3"
```

### Steps to reproduce
Context: I recently received a Laptop from IBM with RHEL 8.3. IBM's RHEL image is peculiar in that pdnsd is installed, which may differ from a typical RHEL installation. pdnsd is a DNS server designed for local caching of DNS information. Unfortunately it silently gets in the way of setting up CodeReady Containers. Because of this, while attempting to install CodeReady Containers you may get warnings related to DNS resolution not working correctly.

  1. Run ```crc setup```
  2. Run ```crc start -p pull-secret.txt```

### Expected
  1. Expected local Openshift cluster to be started started and wildcard on apps-crc.testing to works.
  2. Expected ```crc setup``` to detect pdnsd package is installed, pdnsd service is running, and offer to disable the service or remove the package.
  3. Expected ```crc start``` output **WARN** error to better indicate the cause of the DNS issue.

### Actual
Due to the error below, Web console is accessible but deployed applications are not. 

```bash
$ crc start -p pull-secret.txt
...
WARN Failed public DNS query from the cluster: ssh command error:
command: host -R 3 quay.io
err: Process exited with status 1\n : Host quay.io not found: 2(SERVFAIL)
...
WARN Wildcard DNS resolution for apps-crc.testing does not appear to be working
...
```

Additionally (out of scope of this bug) crc makes changes to DNS resolution server in ```/etc/resolv.conf``` is and adds hosts in ```/etc/hosts``` which do not seem to be restored when running ```crc delete```. This may leave the user unable to resolve domain names after running ```crc setup; crc start; crc stop; crc delete```

### Notes
This is very similar to what was previously reported by arturobrzut in issue #1663 back in Nov 2020, which has since gone stale.

I opened this bug to request that we modify ```crc setup``` to have it perform an additional checks for the pdnsd package and service, thereby guiding the user to resolve the issue. To resolve the issue, see fix below.

### Fix
  1. Disable the conflicting pdnsd service:

```bash
sudo systemctl stop pdnsd.service
sudo systemctl disable pdnsd.service
```

  2. Add the following lines to /etc/NetworkManager/dnsmasq.d/crc.conf in order to use Google's Public DNS servers (8.8.8.8 and 8.8.4.4), as well as DNS provided by your router. You should verify your router address or local DNS server address, in my case it's 192.168.0.1 so I added the following lines:
```bash
server=8.8.8.8
server=8.8.4.4
server=192.168.0.1
```

  3. ```crc start``` will now run without DNS errors.

**Comment by praveenkumar** [comment]
We are working on vsock which will ideally remove those configurations. With 1.26.0 release we are adding doc section around how you can use vsock.

**Comment by cfergeau** [comment]
And even without vsock, I'm not sure we would be adding automatic configuration on non-default RHEL installs. We should detect problematic situations better though (for example, error out early if we detect that a DNS server is running, which means we can't configure dnsmasq correctly).
